client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -267,6 +267,15 @@ private void fetchUpdatesIfNeeded() {
             sync.fetchUpdates();
             long groupRefreshTimeMillis = sync.getState().getConfig().getGroupRefreshTimeMillis();
             fetchStateTimer.reset(Duration.ofMillis(groupRefreshTimeMillis));
+            compactIfNeeded();
+        }
+    }
+    
+    private void compactIfNeeded() {
+        ReaderGroupState state = sync.getState();
+        //Make sure it has been a while, and compaction are staggered.
+        if (state.getUpdatesSinceCompaction() > 500 && Math.random() < 0.05) {
```

Comment:
can we at least extract these numbers in some class-level constants, so it's easier to change them later?